src/formatDate.js

```diff
@@ -50,7 +50,50 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateArray = date.split(fromFormat[3]);
+  const newDate = [];
+
+  for (let i = 0; i < 3; i++) {
+    const part = toFormat[i];
+    const index = fromFormat.indexOf(part);
+
+    newDate[i] = dateArray[index];
+
+    if (!newDate[i]) {
+      newDate[i] = getYear(dateArray, fromFormat, toFormat);
+    }
```

Comment:
You assigned value to array element and then change it. This solution looks strange

Reply:
Let's check if this value need to be converted and assign value only once (you can use `continue` if you need it)